typescript/src/schema/proto.ts

```diff
@@ -586,12 +586,16 @@ function augmentMethod(
 
 export function getSingleHeaderParam(
   rule: protos.google.api.IHttpRule
-): string[] {
+): string[][] {
   const message =
     rule.post || rule.delete || rule.get || rule.put || rule.patch;
   if (message) {
-    const res = message.match(/{(.*?)[=}]/);
-    return res?.[1] ? res[1].split('.') : [];
+    const result: string[][] = [];
+    const matches = message.matchAll(/{(.*?)[=}]/g);
```

Comment:
note: `matchAll` appears in Node.js v12 so you need to update `engines` filed in `package.json` and drop Node 10 support for the generator to be able to safely use it. That's OK, since it's just a generator dependency, the client libraries are not affected by this.

Reply:
Thanks for catching this up. I certainly upgrade to v12 in generator.